apiserver/facades/controller/crosscontroller/package_test.go

```diff
@@ -1,14 +1,50 @@
 // Copyright 2017 Canonical Ltd.
 // Licensed under the AGPLv3, see LICENCE file for details.
 
-package crosscontroller_test
+package crosscontroller
 
 import (
 	"testing"
 
 	gc "gopkg.in/check.v1"
+	"gopkg.in/tomb.v2"
 )
 
 func TestAll(t *testing.T) {
 	gc.TestingT(t)
 }
+
+type mockNotifyWatcher struct {
+	tomb    tomb.Tomb
+	changes chan struct{}
+}
+
+func newMockNotifyWatcher() *mockNotifyWatcher {
+	w := &mockNotifyWatcher{changes: make(chan struct{}, 1)}
+	w.tomb.Go(func() error {
+		<-w.tomb.Dying()
+		return nil
+	})
+	return w
+}
+
+func (w *mockNotifyWatcher) Stop() error {
+	w.Kill()
+	return w.Wait()
+}
+
+func (w *mockNotifyWatcher) Wait() error {
+	return w.tomb.Wait()
+}
+
+func (w *mockNotifyWatcher) Kill() {
+	w.tomb.Kill(nil)
+}
+
+func (w *mockNotifyWatcher) Err() error {
+	return w.tomb.Err()
+}
+
+func (w *mockNotifyWatcher) Changes() <-chan struct{} {
+	return w.changes
+}
```

Comment:
This can be dropped in favour of 

https://github.com/juju/juju/blob/b9cf6c8fdf908cf9b526e7b39a29049dbc61e077/core/watcher/watchertest/notify.go#L17-L56